### PR TITLE
[WIP] Huna/health

### PR DESCRIPTION
**Pull request by Hunachi**
OAuth 同意画面での設定に失敗しているか、（Manifestのintent-filterが）おかしいです。データが正しく取れません。

### CODE REVIEW

**Comment by Hunachi** [comment]
@fumiya-kume 今更ですが、多分治りました。

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/GoogleFitApiDebugActivity.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
+package jp.hacks.smartbread
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.renderscript.Element
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/GoogleFitApiDebugActivity.kt:7]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/GoogleFitApiDebugActivity.kt#L7"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/MainActivity.kt [hunk 2]
```diff
@@ -24,6 +24,8 @@ class MainActivity : AppCompatActivity() {
                     .commitNow()
             }
         }
+
+        GoogleFitApiDebugActivity.startActivity(this)
     }
 
     fun navigateToFaceFragment(){
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/MainActivity.kt:31]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/MainActivity.kt#L31"></span>Missing spacing before “{“</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.ViewModelProviders
-import java.util.Locale
+import com.google.android.gms.auth.api.signin.GoogleSignIn
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:13]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L13"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 4]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
...
+import com.google.android.gms.fitness.Fitness
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:14]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L14"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 5]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
...
+import com.google.android.gms.fitness.FitnessOptions
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:15]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L15"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 6]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
...
+import com.google.android.gms.fitness.data.DataType
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:16]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L16"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 7]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
...
+import com.google.android.gms.fitness.request.DataReadRequest
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:17]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L17"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 8]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
...
+import com.google.android.gms.tasks.OnFailureListener
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:18]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L18"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 9]
```diff
@@ -10,13 +10,20 @@ import androidx.camera.core.CameraX
...
+import com.google.android.gms.tasks.OnSuccessListener
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:19]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L19"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 10]
```diff
@@ -71,4 +78,6 @@ class MainFragment : Fragment() {
 
         viewModel.startTimer(this.requireContext())
     }
+
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:81]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L81"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt [hunk 11]
```diff
@@ -71,4 +78,6 @@ class MainFragment : Fragment() {
...
+
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt:82]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L82"></span>Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

File: android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/health/HealthPermissionUseCaseImpl.kt [hunk 12]
```diff
@@ -0,0 +1,33 @@
+package jp.hacks.smartbread.ui.main.health
+
+import android.content.Context
```
Comment (fumiya-kume) [inline android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/health/HealthPermissionUseCaseImpl.kt:3]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/health/HealthPermissionUseCaseImpl.kt#L3"></span>Unused import</td>
    </tr>
  </tbody>
</table>

**Comment by fumiya-kume** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L3">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/MainFragment.kt#L3</a> - Imports must be ordered in lexicographic order without any empty lines in-between</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/jphacks/FK_1908/blob/4be4dc672423d5ef49ebf08343c4672595c380d0/android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/tts/TTSServiceImpl.kt#L5">android/Smartbread/app/src/main/java/jp/hacks/smartbread/ui/main/tts/TTSServiceImpl.kt#L5</a> - Wildcard import (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**